src/cheri/insns/jalr_32bit.adoc

```diff
@@ -16,56 +16,48 @@ include::base_isa_extension.adoc[]
 
 #Begin changed since last ARC review#
 
+
 Description::
 Indirect jump to the target capability in `{cs1}` with an address offset.
 +
+NOTE: The  description below contains three hooks for extending <<JALR_CHERI>> behavior, used by extensions and/or encoding formats to give a large degree of extensibility.
+ The simpler operation for <<rv32y_cap_description>>/<<rv64y_cap_description>> only uses hook 2.
+ Examples of how CHERIoT encoding formats uses all of the hooks is included below the description.
+ See the full details of the CHERIoT encoding formats for the full specifications, for example: <<rv32y_cheriot_encoding1_description>>.
++
 Operationally, copy `{cs1}` to the target <<pcc>> and then...
 +
 . Compute the target <<pcc>> address in two steps, each using the semantics of the <<SCADDR>> instruction:
 .. Increment the address of the target <<pcc>> by the sign-extended 12-bit `offset`, and
 .. set bit zero of the target <<pcc>> address to zero.
-. Clear the {ctag} of the target <<pcc>> if the triple of its <<sec_cap_type>> value, `{cd}`, and `{cs1}` is not acceptable, as defined by the capability encoding.
-+
-NOTE: All such triples are acceptable to <<rv64y_cap_description>> and <<rv32y_cap_description>>.
-. If the target <<pcc>> is a <<sentry_cap>> (for <<rv32y_cap_description>>/<<rv64y_cap_description>>: if it has <<sec_cap_type>> `1`), then unseal it.
+. HOOK 1: Clear the {ctag} of the target <<pcc>> if the triple of its <<sec_cap_type>> value, `{cd}`, and `{cs1}` is not acceptable, as defined by the capability encoding.
+. If the target <<pcc>> is a <<sentry_cap>>, then unseal it.
 . If the target <<pcc>> is (still) sealed, clear its {ctag}.
-+
-NOTE: This is not possible for <<rv32y_cap_description>>/<<rv64y_cap_description>>, given its two <<sec_cap_type>> values.
 . If `{cd}≠{creg}0`, compute the return capability and install it to `{cd}`:
 .. add the width of this instruction to the current <<pcc>>, and
-.. if the capability encoding defines <<sentry_cap>> <<sec_cap_type>> values for return pointers, then seal the return capability as specified.
-+
-NOTE: <<rv32y_cap_description>>/<<rv64y_cap_description>> always seals return capabilities with <<sec_cap_type>> `1`.
+.. HOOK 2: if the capability encoding defines <<sentry_cap>> <<sec_cap_type>> values for return pointers, then seal the return capability as specified.
```

Comment:
```suggestion
.. HOOK 2: if the capability encoding or an extension defines <<sentry_cap>> <<sec_cap_type>> values for return pointers, then seal the return capability as specified.
```

Reply:
I wonder if this should be extension again? Then it could be dependent on Zys/Zycheriot/Zycfi?